nix/third-party/gRPC-haskell/core/src/Network/GRPC/LowLevel/Server/Unregistered.hs

```diff
@@ -44,13 +47,13 @@ withServerCall s f =
 withServerCallAsync :: Server
                     -> (ServerCall -> IO ())
                     -> IO ()
-withServerCallAsync s f =
+withServerCallAsync s f = mask $ \unmask ->
   serverCreateCall s >>= \case
     Left e -> do grpcDebug $ "withServerCallAsync: call error: " ++ show e
                  return ()
     Right c -> do wasForkSuccess <- forkServer s handler
```

Comment:
Double checked, and `forkServer` will inherit masking state in the forked thread, so this should do the right thing.

Reply:
Heh, I also always have to check the docs for this :slightly_smiling_face: